development/tools/pkg/longlastingdnscleaner/dns_wrapper.go

```diff
@@ -0,0 +1,55 @@
+package dnscleaner
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	dns "google.golang.org/api/dns/v1"
+)
+
+// DNSAPIWrapper abstracts GCP DNS Service API
+type DNSAPIWrapper struct {
+	Context context.Context
+	Service *dns.Service
+}
+
+// LookupDNSEntry delegates to DNS.Service.ResourceRecordSets.List(project, zone, name, address, recordType, recordTTL) function
+func (daw *DNSAPIWrapper) LookupDNSEntry(project, zone, name, address, recordType string, recordTTL int64) (*dns.ResourceRecordSet, bool, error) {
```

Comment:
Return just `(*dns.ResourceRecordSet, error)`. The bool is misleading. For example in some cases in the future you will also want to retry, so wrapper shouldn't decide it - logic should be based on error type.